tinygrad/ops.py

```diff
@@ -46,7 +46,7 @@ def __eq__(self, __value: object) -> bool:
     return self.op == __value.op and self.src == __value.src and self.arg == __value.arg
   def __hash__(self) -> int: return hash((self.op, self.src, self.arg))
   @property
-  def key(self): return (self.op, tuple(map(lambda x: getattr(x, "key", x), self.src)), getattr(self.arg, "key", self.arg))
+  def key(self): return (self.op, tuple(getattr(x, "key", x) for x in self.src), getattr(self.arg, "key", self.arg))
```

Comment:
How does a list comprehension perform here? :)

Reply:
Good call, list comprehension seems to actually speed this up slightly over both `map` and the generator (which are about the same: https://stackoverflow.com/a/1247490/5465108).

Microbenchmark:

```python
import timeit

class T:
    key = 1

a = (T(), T(), T(), T())
N = 1_000_000

lstc = timeit.timeit("tuple([getattr(x, 'key', x) for x in a])", number=N, globals=globals())
genc = timeit.timeit("tuple(getattr(x, 'key', x) for x in a)", number=N, globals=globals())
lamb = timeit.timeit("tuple(map(lambda x: getattr(x, 'key', x), a))", number=N, globals=globals())

print(f"list comp: {lstc:.4f}")
print(f"generator: {genc:.4f}")
print(f"lambda:    {lamb:.4f}")
```
```
list comp: 0.2559
generator: 0.3491
lambda:    0.3633
```